core/corehttp/logs.go

```diff
@@ -0,0 +1,42 @@
+package corehttp
+
+import (
+	"io"
+	"net/http"
+
+	core "github.com/ipfs/go-ipfs/core"
+	"github.com/ipfs/go-ipfs/thirdparty/eventlog"
+)
+
+type writeErrNotifier struct {
+	w    io.Writer
+	errs chan error
+}
+
+func newWriteErrNotifier(w io.Writer) (io.Writer, <-chan error) {
+	ch := make(chan error, 1)
+	return &writeErrNotifier{
+		w:    w,
+		errs: ch,
+	}, ch
+}
+
+func (w *writeErrNotifier) Write(b []byte) (int, error) {
+	n, err := w.w.Write(b)
+	if err != nil {
+		w.errs <- err
+	}
+	return n, err
+}
+
+func LogOption() ServeOption {
+	return func(n *core.IpfsNode, mux *http.ServeMux) (*http.ServeMux, error) {
+		mux.HandleFunc("/logs", func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			wnf, errs := newWriteErrNotifier(w)
+			eventlog.WriterGroup.AddWriter(wnf)
+			<-errs
```

Comment:
if worried about perf (which we may be, given that this is _per event_ (and there are lots...), another option is:

```go
// in Write()

// write to all writers, and nil out the broken ones.
for i, w := range mw.writers {
  _, err := w.Write(b)
  if err != nil {
    mw.writers[i] = nil
  }
}

// consolidate the slice
for i := 0; i < len(mw.writers); i++ {
  if mw.writers[i] != nil {
    continue
  }
  
  j := len(mw.writers)
  for  ; j > i; j-- {
    if mw.writers[j] != nil {
      mw.writers[i], mw.writers[j] = mw.writers[j], nil // swap
      break
    }
  }
  mw.writers = mw.writers[:j]
}
```
seem to work: http://play.golang.org/p/AV9UeZphgq